### PR TITLE
[YandexVideoPreview] Add extractor for yandex video preview urls

### PR DESCRIPTION
**Pull request by KiberInfinity**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [ ] Improvement
- [x] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

Explanation of your *pull request* in arbitrary form goes here. Please make sure the description explains the purpose and effect of your *pull request* and is worded well enough to be understood. Provide as much context and examples as possible.

Support for videos from yandex video search page. (#1794)
URL examples:
https://yandex.ru/video/preview/?filmId=10682852472978372885&text=summer
https://yandex.ru/video/preview/5275069442094787341

### CODE REVIEW

File: yt_dlp/extractor/yandexvideo.py [hunk 1]
```diff
@@ -148,6 +148,40 @@ def _real_extract(self, url):
...
 
 
+class YandexVideoPreviewIE(InfoExtractor):
+    _VALID_URL = r'''(?x)
+                    https?://
+                        (?:
+                            yandex\.ru/video/preview(?:/?\?.*?filmId=|/)
+                        )
+                        (?P<id>\d+)
+                    '''
```
Comment (Ashish0804) [inline yt_dlp/extractor/yandexvideo.py:158]: proly not the best but this is more compact and what i prefer tbh
```suggestion
    _VALID_URL = r'https?://(?:www\.)?yandex\.ru/video/preview/(?:\?.*filmId=)?(?P<id>\d+)'
```

File: yt_dlp/extractor/yandexvideo.py [hunk 2]
```diff
@@ -148,6 +148,40 @@ def _real_extract(self, url):
...
+    }, {  # YandexVideo
+        'url': 'https://yandex.ru/video/preview/5275069442094787341',
+        'only_matching': True,
+    }, {  # youtube
+        'url': 'https://yandex.ru/video/preview/?filmId=16658118429797832897&from=tabbar&p=1&text=%D0%BF%D1%80%D0%BE%D1%81%D0%BC%D0%BE%D1%82%D1%80+%D1%84%D1%80%D0%B0%D0%B3%D0%BC%D0%B5%D0%BD%D1%82%D0%B0+%D0%BC%D0%B0%D0%BB%D0%B5%D0%BD%D1%8C%D0%BA%D0%B8%D0%B9+%D0%BF%D1%80%D0%B8%D0%BD%D1%86+%D0%BC%D1%8B+%D0%B2+%D0%BE%D1%82%D0%B2%D0%B5%D1%82%D0%B5+%D0%B7%D0%B0+%D1%82%D0%B5%D1%85+%D0%BA%D0%BE%D0%B3%D0%BE+%D0%BF%D1%80%D0%B8%D1%80%D1%83%D1%87%D0%B8%D0%BB%D0%B8',
+        'only_matching': True,
+    }, {  # Odnoklassniki
+        'url': 'https://yandex.ru/video/preview/?text=Francis%20Lai%20-%20Le%20Bon%20Et%20Les%20MC)chants&path=wizard&parent-reqid=1643208087979310-1481782809207673478-sas3-0931-2f9-sas-l7-balancer-8080-BAL-9380&wiz_type=vital&filmId=12508152936505397283',
+        'only_matching': True,
+    }]
```
Comment (Ashish0804) [inline yt_dlp/extractor/yandexvideo.py:174]: Add info or atleast one of the test

File: yt_dlp/extractor/yandexvideo.py [hunk 3]
```diff
@@ -148,6 +148,40 @@ def _real_extract(self, url):
...
+
+    def _real_extract(self, url):
+        id = self._match_id(url)
+        webpage = self._download_webpage(url, id)
+        data_raw = re.findall(r'window.Ya.__inline_params__\s*=\s*JSON.parse\(\'([^"]+?\\u0022video\\u0022:[^"]+?})\'\);', webpage)[-1].encode().decode("unicode-escape")
+        data_json = self._parse_json(data_raw, id)
+        video_page_url = try_get(data_json, lambda x: x['video']['url'])
+        return self.url_result(video_page_url)
```
Comment (Ashish0804) [inline yt_dlp/extractor/yandexvideo.py:182]:
```suggestion
        return self.url_result(data_json['video']['url'])
```

File: yt_dlp/extractor/yandexvideo.py [hunk 4]
```diff
@@ -148,6 +148,40 @@ def _real_extract(self, url):
...
+    }, {  # Odnoklassniki
+        'url': 'https://yandex.ru/video/preview/?text=Francis%20Lai%20-%20Le%20Bon%20Et%20Les%20MC)chants&path=wizard&parent-reqid=1643208087979310-1481782809207673478-sas3-0931-2f9-sas-l7-balancer-8080-BAL-9380&wiz_type=vital&filmId=12508152936505397283',
+        'only_matching': True,
+    }]
+
+    def _real_extract(self, url):
+        id = self._match_id(url)
+        webpage = self._download_webpage(url, id)
+        data_raw = re.findall(r'window.Ya.__inline_params__\s*=\s*JSON.parse\(\'([^"]+?\\u0022video\\u0022:[^"]+?})\'\);', webpage)[-1].encode().decode("unicode-escape")
+        data_json = self._parse_json(data_raw, id)
```
Comment (pukkandan) [inline yt_dlp/extractor/yandexvideo.py:180]: Use `self._search_regex` and pass any decoding function to the `transform` of `_parse_json`

**Review by pukkandan — CHANGES_REQUESTED** [review]

File: yt_dlp/extractor/yandexvideo.py [hunk 5]
```diff
@@ -149,16 +150,22 @@ def _real_extract(self, url):
...
 
 class YandexVideoPreviewIE(InfoExtractor):
-    _VALID_URL = r'''(?x)
-                    https?://
-                        (?:
-                            yandex\.ru/video/preview(?:/?\?.*?filmId=|/)
-                        )
-                        (?P<id>\d+)
-                    '''
+    _VALID_URL = r'https?://(?:www\.)?yandex\.ru/video/preview/?(?:\?.*?filmId=|)(?P<id>\d+)'
```
Comment (pukkandan) [inline yt_dlp/extractor/yandexvideo.py:153]:
```suggestion
    _VALID_URL = r'https?://(?:www\.)?yandex\.ru/video/preview(?:/?\?.*?filmId=|/)(?P<id>\d+)'
```
This is more correct, no?

File: yt_dlp/extractor/yandexvideo.py [hunk 6]
```diff
@@ -148,6 +149,45 @@ def _real_extract(self, url):
...
+        'only_matching': True,
+    }, {  # Odnoklassniki
+        'url': 'https://yandex.ru/video/preview/?text=Francis%20Lai%20-%20Le%20Bon%20Et%20Les%20MC)chants&path=wizard&parent-reqid=1643208087979310-1481782809207673478-sas3-0931-2f9-sas-l7-balancer-8080-BAL-9380&wiz_type=vital&filmId=12508152936505397283',
+        'only_matching': True,
+    }]
+
+    def _real_extract(self, url):
+        id = self._match_id(url)
+        webpage = self._download_webpage(url, id)
+        data_raw = self._search_regex(r'window.Ya.__inline_params__\s*=\s*JSON.parse\(\'([^"]+?\\u0022video\\u0022:[^"]+?})\'\);', webpage, "data_raw")
```
Comment (pukkandan) [inline yt_dlp/extractor/yandexvideo.py:186]:
```suggestion
        data_raw = self._search_regex(r'window.Ya.__inline_params__\s*=\s*JSON.parse\(\'([^"]+?\\u0022video\\u0022:[^"]+?})\'\);', webpage, 'data_raw')
```

**Pull request merged**